### PR TITLE
Make chemprot apply consistently to touch reactions

### PR DESCRIPTION
**Pull request by Kubius**
[MAJOR]
## About the PR 

Implements the following set of changes with the aim of making chemprot consistently factor into all touch applications of chemicals (as opposed to only factoring into the extent of skin penetration for skin penetrating chems).

**Main Implementation**

- Adds a reaction_mob_chemprot_layer proc that handles the application of chemprot to the practical volume of a reagent that's applied to a mob, also providing the pre-mitigation volume (raw_volume) so reactions can use it if they logically shouldn't care about hitting a mob's suit instead of the skin itself.
- Places aforementioned proc in other locations where reaction_mob was called without a specified non-touch method.
- Removes the separate chemprot check for skin-penetrating chemicals, since the blocking factor by that point is now your skin instead of your clothing.
- Removes a set of checks in /mob/living/carbon/human/EnteredFluid that would cause reactions to simply not happen at all if specific pieces of gear met certain criteria, meaning general chemprot never even factored in.

**Accommodation Changes**

- Adds an "ignore_chemprot" reaction parameter, causing the chemprot check to be entirely skipped.
- Introduces the "ignore_chemprot" reaction parameter to patches, auto-menders and skin processes, so their efficacy isn't affected by this change.
- Water splashed on a mob now uses raw_volume to determine its effectiveness at extinguishing fires.
- Holy water applied to vampires uses raw_volume to determine its damage, in accordance with the principles of sanctification.
- Glue now uses raw_volume for its determination of stickiness.
- Nitroglycerin now uses raw_volume for its explosion strength as applied to mobs.
- Acetone now uses raw_volume for sticker removal.
- Napalm, chlorine trifluoride and black powder now use raw_volume for contact effect strength / thresholds.
- Fluorosulfuric acid now bases its exposed-to-face damage on raw_volume when 10 or more units are applied, and additionally won't melt face gear whose chemprot exceeds 1/3 of the applied volume. Roughly preserves current behavior; a more in-depth rework would probably be cool, but that's getting into scope creep.

## Why's this needed? 

Chemprot currently has no bearing on topical effects from chems, which is a major lapse in coverage; this fixes that in a way that shouldn't reduce design space for reagent reactions.

Should not change the behavior of *most* individual chemicals in the strictest sense; the change to the applied amount may have other consequences, but I think I've accounted for the most crucial ones (I inspected every reaction_mob for usage of volume).

Whether patches and automenders should be able to go through chemically-protective attire is debatable, but leaving them operational seems like the better option for gameplay purposes.

## Changelog

```changelog
(u)Kubius
(*)Chem protection on gear now properly factors into most chemical touch effects. Menders and patches are exempt. 
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
Not entirely sure if I'm sold on the concept, but at the very least, incendiaries (CLF3, napalm, etc) should probably use raw volume, and acids (notably pacid) should probably use raw volume for the purposes of melting headwear

**Comment by Kubius** [comment]
Adjusted as requested. Sulfuric (plus derivatives) and acetic acids don't melt gear and should theoretically not care that much about raw_volume as a consequence.

**Review by Tarmunora — APPROVED** [review]

**Comment by flrsh** [comment]
All the different accommodation changes make this feel like this change isn't actually applying that consistently.

**Comment by Kubius** [comment]
The goal is "consistently with how you'd think it'd behave to have chems splashed on you", moreso than exactly the same from chem to chem; I perhaps chose the wrong word? The old behavior was very consistent in the sense of not working for literally anything but skin-penetration extent of skin-penetrating chems, but I think that's also actively contrary to how one would expect it to behave.

**Comment by Tarmunora** [comment]
> All the different accommodation changes make this feel like this change isn't actually applying that consistently.

hmm, yeah, the number of things that A.) scale based off of volume on a TOUCH reaction and
B.) aren't just going to use raw volume anyways is... pretty small - more or less just acids and topical medicines

**Comment by Kubius** [comment]
Chemprot check on individual harmful chems instead?

File: code/modules/fluids/fluid_core.dm [hunk 1]
```diff
@@ -810,7 +810,7 @@ var/mutable_appearance/fluid_ma
 			//	if (!M.anchored)
 			//		F.add_tracked_blood(M)
 
-
+/*
```
[the remark this answers is not in the document]
Comment (Kubius) [reply code/modules/fluids/fluid_core.dm]: Reasonable enough. Done in 3f60de6

**Pull request merged**